addons/web/static/tests/legacy/views/kanban_tests.js

```diff
@@ -2794,6 +2794,43 @@ QUnit.module('Views', {
         kanban.destroy();
     });
 
+    QUnit.test('Open record when clicking on widget field', async function (assert) {
+        assert.expect(2);
+
+        var kanban = await createView({
+            View: KanbanView,
+            model: 'partner',
+            data: this.data,
+            arch: '<kanban class="o_kanban_test">' +
+                        '<templates>' +
+                            '<t t-name="kanban-box">' +
+                                '<div class="oe_kanban_global_click">' +
+                                    '<field name="salary" widget="monetary"/>' +
+                                    '<div>' +
+                                        '<a class="o_test_link" href="#">test link</a>' +
+                                    '</div>' +
+                                '</div>' +
+                            '</t>' +
+                        '</templates>' +
+                    '</kanban>',
+            archs: {
+                'product,false,form': '<form string="Product"><field name="display_name"/></form>',
+            },
+            intercepts: {
+                switch_view: function (ev) {
+                    assert.deepEqual(_.pick(ev.data, 'res_id', 'view_type'), {
+                        res_id: 1,
+                        view_type: 'form',
+                    }, "should trigger an event to open the form view");
+                },
+            },
+        });
+        assert.containsN(kanban, '.o_kanban_record:not(.o_kanban_ghost)', 4);
+        const $salary = kanban.$(".oe_kanban_global_click .o_field_monetary[name=salary]:eq(0)");
+        $salary.click();
```

Comment:
Since we only use the `$salary` variable to do a click, we could do it like this:
```suggestion
        kanban.$(".oe_kanban_global_click .o_field_monetary[name=salary]:eq(0)")
              .click();
```